packages/server-wallet/src/wallet/store.ts

```diff
@@ -43,13 +43,33 @@ const throwMissingChannel: MissingAppHandler<any> = (channelId: string) => {
   throw new ChannelError(ChannelError.reasons.channelMissing, {channelId});
 };
 
-export const Store = recordFunctionMetrics({
-  getFirstParticipant: async function(knex: Knex): Promise<Participant> {
-    const signingAddress = await Store.getOrCreateSigningAddress(knex);
-    return {participantId: signingAddress, signingAddress, destination: makeDestination(HashZero)};
-  },
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export class Store {
+  constructor(readonly timingMetrics: boolean) {
```

Comment:
Now that Store is a class, it can be constructed with a knex attribute, an, meaning we wouldn't need to pass around the knex object.

Reply:
Yeah that's a possible route -- you wouldn't need to pass around the `knex` object inside the `Store` class, but you would still need to accept transactions in the `Store` methods, and have them fall back to `this.knex` if no `tx` is provided.

Also the `Wallet` class does stuff with the `knex` object that totally bypasses the `Store`. So it seems right that the `Wallet` owns the `knex` object. I guess it seems marginal to me whether it should be passed in to the `Store` constructor, or each method. We could consider it in a future PR?